### DISCUSSION TITLE
Problem with Named Version Checkpoint API

### DISCUSSION THREAD

**Discussion by youngbrett** [iTwin/itwinjs-core#4311] [Q&A] [answered]
Hello --

I'm using the "Try" window in the documentation for the [Get Named Version Checkpoint](https://developer.bentley.com/apis/imodels/operations/get-named-version-checkpoint/) operation and I can't get it to work.  I previously used it and it was helpful, but now I'm getting an error.  I can view the Project, iModel, and Named Version, for which I'm the owner, and I am using an authentication token.  I believe the request is finding the iModel, but failing on accessing the Named Version because I get the following error:

"code": "NamedVersionNotFound",  "message": "Requested Named Version is not available."

If I wasn't accessing the iModel, it would say the iModel isn't found.  I'm getting the same behavior across a few projects and iModels, some of which are shared.

Any help?  I don't think I changed my process.

(Great conference today, btw.)

**Comment by calebmshafer** [comment, marked as answer]
Hey @youngbrett, would you be able to share the `x-correlation-id` of the request that is returning that response?

cc: @robertas-kerpys

Glad you enjoyed the conference!

**Reply by youngbrett** [reply]
7bfbc934-d9b7-43ae-9748-22711aca9f77

**Reply by youngbrett** [reply]
Similarity [here](https://github.com/iTwin/itwinjs-core/discussions/4087)?

**Reply by calebmshafer** [reply]
Doesn't appear to be the same issue as the one you linked since your request is making it to the service and returning a response. I just checked and it looks like the iModel should have two named versions but for some reason is returning them correctly.  The team is looking into this now to see why that is the case.

**Reply by youngbrett** [reply]
This response happened across a variety of Projects, iModels, and Named Versions.  Let me know if more x-correlation-id's would help.  Thanks for your attention to this.

**Reply by calebmshafer** [reply]
Yes, if you have some more that would be great. This will help us narrow down the issue. Thanks!

**Reply by calebmshafer** [reply]
@youngbrett can you try getting the list of Named Versions for that iModel to ensure that's working correctly using this API https://developer.bentley.com/apis/imodels/operations/get-imodel-named-versions/? If that works, would you mind posting the ids for the Named Versions that are returned?

**Reply by youngbrett** [reply]
**Here goes with data for three projects.  For project1, there's the result of Get Named Versions:**

{
    "namedVersions": [{
        "id": "1280d8a9-bb27-495b-b563-70e0290ac6e3",
        "displayName": "Test1",
        "changesetId": "552df09e56cbf69efdee2e183581eafa9073c8c0",
        "changesetIndex": 32
    }],
    "_links": {
        "self": {
            "href": "https://api.bentley.com/imodels/a6f70f75-1dfe-44db-8ff8-209a979a2101/namedversions?$skip=0&$top=100"
        },
        "prev": null,
        "next": null
    }
}

Get Named Version Checkpoint for ChangeSetID 552df09e56cbf69efdee2e183581eafa9073c8c0  results error with x-correlation-id of 7235accc-0e4c-4074-b4c5-03ecd7359015.

**Project 2 has the following Get Named Version response:**  

{
    "namedVersions": [{
        "id": "48be6901-eceb-4a9c-9e2c-5bf2b84a81d3",
        "displayName": "Warehouse_v4_NamedVersion_v1",
        "changesetId": "98fc3233c3ea256cfa0d8c866acbb0bb4b408e50",
        "changesetIndex": 5
    }, {
        "id": "7d33d719-33cb-45a5-b887-baf77244dd3a",
        "displayName": "Warehouse_v4_NamedVersion_v2",
        "changesetId": "c9ea2fe7fc0fd82799e7a54a33eb1e6c773554b5",
        "changesetIndex": 4
    }],
    "_links": {
        "self": {
            "href": "https://api.bentley.com/imodels/d0cac6ab-d92a-48f8-9e03-aca7379af640/namedversions?$skip=0&$top=100"
        },
        "prev": null,
        "next": null
    }
}

And the x-correlation-id for ChangeSetID 98fc3233c3ea256cfa0d8c866acbb0bb4b408e50 is 3d8139a2-da51-4078-b34a-0525264a79d1.  In addition, see attached screenshots for this project's GetNamedVersionCheckPoint fields.  
![Screen Shot 2022-09-20 at 5 58 28 PM](https://user-images.githubusercontent.com/47938388/191373236-a5449d4e-879c-4860-b6a5-edcb176aa1e2.png)
![Screen Shot 2022-09-20 at 5 58 11 PM](https://user-images.githubusercontent.com/47938388/191373239-2cbbaddd-f072-4927-9da8-22dc70f0ad66.png)

**A third project which is shared with Tim Lawrence, Al Anora, Matt Gooding, and Jeff Marker has the following info:**

NamedVersionID: 40c23cc5-113d-4637-959b-fe6fc45d6387
ChangeSetID:  43aabd860db1545e52f46f1615706695cc905ca0
X-Correlation-ID:  69d69b64-bd65-4af9-b1f9-3ba30ec36422

**Reply by calebmshafer** [reply]
Would you be able to try putting the `id` that is returned as part of the Get Named Versions request in as the `namedVersionId` parameter instead of the changesetId?  So for **Project 2** the `namedVersionId` would be 48be6901-eceb-4a9c-9e2c-5bf2b84a81d3 for the changesetId of 98fc3233c3ea256cfa0d8c866acbb0bb4b408e50.

**Reply by youngbrett** [reply]
Error - IModelNotFound.  x-correlation-id: f4273894-ab1a-415c-970b-3e890ed440f6

![Screen Shot 2022-09-20 at 9 41 13 PM](https://user-images.githubusercontent.com/47938388/191395300-e60fa326-79e8-4974-92d2-7e1eb4e17958.png)
![Screen Shot 2022-09-20 at 9 41 00 PM](https://user-images.githubusercontent.com/47938388/191395304-9bbe9dc9-ec54-42f2-b739-486e46a88ae9.png)

**Reply by calebmshafer** [reply]
I think if you switch around `id` to be `d0cac6ab-d92a-48f8-9e03-aca7379af640` and then `namedVersionId` to be `48be6901-eceb-4a9c-9e2c-5bf2b84a81d3` that should work.

**Reply by youngbrett** [reply]
Yup - that did it!  Thanks for the help.  

Here's where my head got lost.  I thought a ChangeSetID was the GUID for the NamedVersion.  I think I confused myself  because I was deconstructing URL's to get my parameters and Bentley's developer site uses ChangeSetIDs whenever you ask for a NamedVersion through a link or viewing it as a model. When I wanted to request a Named Version, I assumed they were the same.  My bad and thanks for the help!  

Bentley should consider adding NamedVersionID as an information property or detail in iModelHub, rather than getting it through the API.  That would make it clear they are different and I think it's tough to find.
